docs/time-aware-issuers.md

```diff
@@ -0,0 +1,90 @@
+# Time Aware Issuers
+
+There exists a need to have issuers support different signing keys for configurable
+intervals of times, whereby an issuer can sign with a particular key representing a
+finite interval of time, and on that period of time the signature will be valid for
+redemption.
+
+## Database Changes Required
+
+To support this feature, the following new tables will need to be created:
+
+```sql
+-- time_aware_issuer will represent the v3 issuer, which has time aware signing keys
+create table time_aware_issuer(
+    id uuid primary key default generate_uuidv4(),
+    issuer_type text not null,
+    created_at datetime with time zone,
+    expires_at datetime with time zone,
+    last_rotated_at datetime with time zone,
+    max_tokens integer,
+    buffer integer,
+    duration text
+);
+
+-- time_aware_keys will represent the v3 issuer keys which are time aware
+create table time_aware_keys (
+    id uuid primary key default generate_uuidv4(),
+    time_aware_issuer_id uuid references time_aware_issuer(id),
+    created_at datetime with time zone,
+    start_at datetime with time zone,
+    end_at datetime with time zone,
+    signing_key text not null,
+    public_key text not null,
```

Comment:
Inclusion of public key here is excellent. Please do not remove it in the course of development of this feature. +1